jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -40,11 +40,14 @@ def put(self, key: str, value: bytes):
     if not key:
       raise ValueError("key cannot be empty")
     if self._evict_entries_if_necessary(key, value):
-      self._create_a_file(self._path, key, value)
+      path_to_new_file = os.path.join(self._path, key)
+      with open(path_to_new_file, "wb") as file:
+        file.write(value)
     else:
-      warnings.warn("This compilation file is larger than the size of the cache")
+      warnings.warn(("Cache value of size {} is larger than the max cache size of {}").format(len(value), self._max_cache_size_bytes))
```

Comment:
You should use a formatted string literal (https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) instead of the `format` function. Also add a line break to keep lines less than about 80 characters long. Like this:

```python
      warnings.warn(f"Cache value of size {len(value)} is larger than the "
                    "max cache size of {self._max_cache_size_bytes}")
```